cdap-distributions/bin/prep_and_stage_bundles.sh

```diff
@@ -0,0 +1,136 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+######################################################################################
+# Create RPM and Debian bundles and upload them directly to docs1
+# This script expects to receive 3 parameters in this order:
+#   $1 BUILD_PACKAGE    e.g. cdap
+#   $2 VERSION    e.g. 3.0.4
+#   $3 USER                     REMOTE_USER
+#   $4 remote host              REMOTE_HOST
+#   $5 target directory         REMOTE_INCOMING_DIR
+######################################################################################
+DEBUG=${DEBUG:-no}              ### set to yes for debugging
+RUN_DATE=`date '+%Y%m%d_%R'`
+SCRIPT=`basename ${BASH_SOURCE[0]}`                     ### Set Script Name variable
+BUILD_PACKAGE="${1:-cdap}"                                    ## e.g. cdap ## a.k.a. BUILD_PACKAGE
+VERSION="${2}" ## e.g. 3.0.1
+REMOTE_USER=${3}                                        ### remote user
+REMOTE_HOST=${4:-127.0.0.1}                             ### remote host
+REMOTE_INCOMING_DIR=${5}                                ### target directory on remote host
+BUILD_PACKAGE=${BUILD_PACKAGE:-cdap}
+REMOTE_BASE_DIR="incoming"
+USER=${USER:-bamboo}
+TMP_DIR='/tmp/bundles'
+S3STG_DIR=${S3STG_DIR:-/data/s3stg} # change it #####
+DEB_BUNDLE="${BUILD_PACKAGE}-deb-bundle"
+RPM_BUNDLE="${BUILD_PACKAGE}-rpm-bundle"
+DEB_BUNDLE_DIR="${TMP_DIR}/${DEB_BUNDLE}"
+RPM_BUNDLE_DIR="${TMP_DIR}/${RPM_BUNDLE}"
+DISTRIBUTED_DEB_BUNDLE="${BUILD_PACKAGE}-distributed-deb-bundle"
+DISTRIBUTED_RPM_BUNDLE="${BUILD_PACKAGE}-distributed-rpm-bundle"
+DEB_BUNDLE_TGZ="${DISTRIBUTED_DEB_BUNDLE}-${VERSION}.tgz"
+RPM_BUNDLE_TGZ="${DISTRIBUTED_RPM_BUNDLE}-${VERSION}.tgz"
+DEB_BUNDLE_TGZ_PATH="${DEB_BUNDLE_DIR}/${DEB_BUNDLE_TGZ}"
+RPM_BUNDLE_TGZ_PATH="${RPM_BUNDLE_DIR}/${RPM_BUNDLE_TGZ}"
+S3STG_INCOMING="${S3STG_DIR}/incoming"
+S3STG_INCOMING_PROJ="${S3STG_INCOMING}/${BUILD_PACKAGE}"
+STG_DIR="${S3STG_INCOMING}/downloads/co/cask/${BUILD_PACKAGE}"
+STG_DEB_DIR="${STG_DIR}/${DISTRIBUTED_DEB_BUNDLE}/${VERSION}"
+STG_RPM_DIR="${STG_DIR}/${DISTRIBUTED_RPM_BUNDLE}/${VERSION}"
+echo ${STG_RPM_DIR}
+
+#############################
+# output trimmer
+decho () {
+  if [[ "${DEBUG}" == 'yes' ]]; then
+    echo ${*}
+  else
+    RSYNC_QUIET='--quiet'
+  fi
+}
+
+# Help function
+function HELP {
+  echo -e \\n"Help documentation for ${BOLD}${SCRIPT}.${NORM}"\\n
+  echo -e "${REV}Basic usage:${NORM} ${BOLD}$SCRIPT${NORM} <remote_user> <remote_host> <remote_target_directory"\\n
```

Comment:
Avoid ANSI code changes. They're horrible on text-only terminals that don't support bold/reverse.